perllib/FixMyStreet/Cobrand/UKCouncils.pm

```diff
@@ -448,17 +455,21 @@ sub _fetch_features {
 
     my $uri = $self->_fetch_features_url($cfg);
     my $response = get($uri) or return;
-
     my $j = JSON->new->utf8->allow_nonref;
     try {
         $j = $j->decode($response);
     } catch {
         # There was either no asset found, or an error with the WFS
         # call - in either case let's just proceed without the USRN.
-        return;
+        # Additionally try for a different map type response
```

Comment:
This is not a good way to do this. If you know that you're going to be requesting XML in fetch_features, it should be passed in as a variable - not trying to decode as JSON and then trying XML if that fails. But a better solution would be to request JSON like all the other uses of this function do, which will need a config line adding to Thamesmead's .map file to say JSON output is allowed.

Reply:
I've worked this through with the current maps which are on Tilma, but there's a PR I'm just waiting for them to confirm which takes four of the five layers to their mapservers. So I guess I'll need to go with passing in a variable... https://github.com/mysociety/fixmystreet/pull/4065

Reply:
Ah, and they might not have JSON output then, okay. You should still add a parameter to fetch_features saying you expect/want XML then, I think, clearer than knowing you're asking for XML but trying to parse it as JSON first.

Reply:
This hopefully now as wanted